Improve base-conversion (#1506)
enkidevs/curriculum: javascript/core/numbers/base-conversion.md

---
## Content

Base conversion for numerical values in JavaScript is easily achievable
with the help of `toString` and `parseInt` functions.

### toString

> `<number>.toString(base)` operates on a number and converts it into a string in given base

Reviewer: ```suggestion
`<number>.toString(base)` operates on a number and converts it into a string in the given base.
```

For example, converting a decimal number to hexadecimal, octal and binary looks something like this:
```javascript
let number = 123
number.toString(16) // "7b"
number.toString(8) // "173"
number.toString(2) // "1111011"
```

> Note that the type of returned valye is string

Reviewer: ```suggestion
Note that the type of returned value is a string.
```

```javascript
let number = 123
typeof number.toString(16) // "string"
```

### parseInt

`parseInt(<string>, <base>)` takes a string and it's base and converts it into a decimal number.

> For example, parsing a hexadecimal, octal and binary number in string form into a decimal

Reviewer: ```suggestion
For example, parsing a hexadecimal, octal and binary number in string form into a decimal:
```

```javascript
parseInt("7b", 16) // 123
parseInt("173", 8) // 123
parseInt("1111011", 2) // 123
```

> The type of returned value is `number`

Reviewer: ```suggestion
The type of returned value is `number`.
```
